### PR TITLE
Only rebuild selected packages (including actionlib)

### PR DESCRIPTION
**Pull request by Tobias-Fischer**

### CODE REVIEW

**Comment by defunctzombie** [comment]
@Tobias-Fischer so what we should do is use this PR (or one like it that we make) here locally within this fork to iterate on getting the build to work?

**Comment by Tobias-Fischer** [comment]
Please merge it in to your fork - then you can continue to iterate in your fork.

File: vinca_linux_64.yaml [hunk 1]
```diff
@@ -37,149 +37,150 @@ skip_existing:
...
-  - handeye
-  - criutils
-  - baldor
-  # - rosmon
-  - catch-ros
-  - rosfmt
+  - actionlib
   - foxglove_bridge
+  - ros-babel-fish
+  - babel-fish-test-msgs
```
Comment (achim-k) [inline vinca_linux_64.yaml]:
```suggestion
  - ros-babel-fish-test-msgs
```

File: vinca_win.yaml [hunk 2]
```diff
@@ -31,79 +31,86 @@ skip_existing:
...
-  - hector-gazebo-plugins
-  - gazebo-ros-control
-  - gazebo-plugins
-  - effort-controllers
-  - velocity-controllers
-  - joy
+  - actionlib
+  - foxglove_bridge
+  - ros-babel-fish
+  - babel-fish-test-msgs
```
Comment (achim-k) [inline vinca_win.yaml]:
```suggestion
  - ros-babel-fish-test-msgs
```

File: vinca_osx_arm64.yaml [hunk 3]
```diff
@@ -34,89 +34,94 @@ packages_select_by_deps:
...
-  - teleop-twist-keyboard
-  - laser-proc
-  - rgbd-launch
-  - rosserial-arduino
-  - rosserial-client
-  - navigation
+  - actionlib
   - foxglove_bridge
+  - ros-babel-fish
+  - babel-fish-test-msgs
```
Comment (achim-k) [inline vinca_osx_arm64.yaml]:
```suggestion
  - ros-babel-fish-test-msgs
```

File: vinca_osx.yaml [hunk 4]
```diff
@@ -29,129 +29,137 @@ skip_existing:
...
-  - tf2_py
-  - tf2_ros
-  - tf2_sensor_msgs
-  - tf2_tools
-  - gps-common
-  - rosbridge_suite
+  - actionlib
   - foxglove_bridge
+  - ros-babel-fish
+  - babel-fish-test-msgs
```
Comment (achim-k) [inline vinca_osx.yaml]:
```suggestion
  - ros-babel-fish-test-msgs
```

File: vinca_linux_aarch64.yaml [hunk 5]
```diff
@@ -28,84 +28,90 @@ skip_existing:
...
-  - robot_localization
-  - perception
-  - simulators
-  - desktop_full
-  - audio-common-msgs
-  - sound-play
+  - actionlib
   - foxglove_bridge
+  - ros-babel-fish
+  - babel-fish-test-msgs
```
Comment (achim-k) [inline vinca_linux_aarch64.yaml]:
```suggestion
  - ros-babel-fish-test-msgs
```

**Comment by achim-k** [comment]
@Tobias-Fischer it fails now with the following dependency error which is normally resolved by running `rosdep install...`. It seems that there is no `rosdep install` step in the CI. Is this intended? 

```
CMake Error at CMakeLists.txt:22 (find_package):
  By not providing "Findwebsocketpp.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "websocketpp", but CMake did not find one.
```

**Comment by Tobias-Fischer** [comment]
The mapping was missing - should work now.

**Comment by achim-k** [comment]
One step further, but now we have the same issues as the humble build with the environment variables not being passed through (https://github.com/RoboStack/ros-humble/pull/49#issuecomment-1443473152)

```
CMake Error at CMakeLists.txt:166 (message):
  ROS_VERSION environment variable must be 1 or 2
```

**Comment by Tobias-Fischer** [comment]
I'm really not sure what's going on. Previously I saw I was patching `ROS_VERSION` out (https://github.com/RoboStack/ros-humble/blob/11771ff55f8ea1469aa11e145fcdcc6c5ad3d22e/patch/ros-humble-lanelet2.patch#L8) which is not very nice I have to admit.

It would be good to fix the underlying issue, but @wolfv and myself have few cycles those days for debugging. The fastest way to get your PR merged in would probably be again the ugly patch, and opening an issue to fix the underlying issue later on. What do you think @achim-k?

**Comment by achim-k** [comment]
> It would be good to fix the underlying issue, but @wolfv and myself have few cycles those days for debugging. The fastest way to get your PR merged in would probably be again the ugly patch, and opening an issue to fix the underlying issue later on. What do you think @achim-k?

I have opened https://github.com/RoboStack/ros-humble/issues/52 to track the underlying issue.
Adding a patch would mean that every new release potentially invalidates that patch, right? I would somewhat prefer fixing the underlying issue in that case (even if it's just setting the necessary environment variables manually)

**Comment by Tobias-Fischer** [comment]
Thanks for opening the issue!

Correct, a new release potentially invalidates the patch. However, we are pretty good in dealing with that - we have literally hundreds of patches for RobosStack's ros-noetic ..

I totally agree that fixing the underlying issue is preferable. The weird thing is that in https://github.com/RoboStack/ros-humble/pull/49 the environment variables are manually set (https://github.com/RoboStack/ros-humble/pull/49/files#diff-069a2e75bc939d92ad32a50dada65893abaf603c1a586b97183ec1ca476bba23R6-R8) but it somehow does not propagate for the cross-compilation platforms. If you have debugging cycles, it would be amazing if you could look into it.

**Comment by Tobias-Fischer** [comment]
Can you please merge this @achim-k ?

**Pull request merged**